aspnetcore/fundamentals/dependency-injection.md

```diff
@@ -146,45 +209,166 @@ Transient lifetime services are created each time they're requested. This lifeti
 Scoped lifetime services are created once per request.
 
 > [!WARNING]
-> If you're using a scoped service in a middleware, inject the service into the `Invoke` or `InvokeAsync` method. Don't inject via constructor injection because it forces the service to behave like a singleton.
+> When using a scoped service in a middleware, inject the service into the `Invoke` or `InvokeAsync` method. Don't inject via constructor injection because it forces the service to behave like a singleton. For more information, see [Middleware](xref:fundamentals/middleware/index).
 
 **Singleton**
 
-Singleton lifetime services are created the first time they're requested (or when `ConfigureServices` is run if you specify an instance there) and then every subsequent request will use the same instance. If your application requires singleton behavior, allowing the services container to manage the service's lifetime is recommended instead of implementing the singleton design pattern and managing your object's lifetime in the class yourself.
+Singleton lifetime services are created the first time they're requested (or when `ConfigureServices` is run if an instance is specified there) and then every subsequent request uses the same instance. If the app requires singleton behavior, allowing the service container to manage the service's lifetime is recommended. Don't implement the singleton design pattern and provide user code to manage the object's lifetime in the class.
```

Comment:
Add a comma before "if an instance is".

Reply:
Singleton lifetime services are created:

* The first time

Or some other way to keep sentence under 20 words.